### PR TITLE
Fix: Bug fixing in Update shopping cart

### PR DESCRIPTION
**Pull request by ruberwa**
#### What does this PR do?
  * Buyer is able to update their cart

#### Description of Task to be completed?
  * A User who logged in is able to add the item to the cart.
  * If the Product is already in the cart, the Quantity will be updated.
  
      * Buyer is able to update the quantity of product in the cart 

#### How should this be manually tested?
  * Clone the branch
  * Go to its path in the terminal run: npm install for Dependencies
  * Create a .env file
  * Add the following Variables in that .env file
    - JWT_SECRET
    - USER_SECRET_KEY
    - BCRYPT_KEY
    - TOKEN_EXPIRATION
    - SENDGRID_API_KEY
    - SENDER_EMAIL
    - GOOGLE_CLIENT_ID
    - GOOGLE_CLIENT_SECRET
    - NODE_ENV
    - DB_USERNAME
    - DB_PASSWORD
    - DB_HOST
    - DB_NAME
    - RESET_URL

  * Go to this URL to test functionalities: http://localhost:5000/api-docs/
  
        - and you will see a tag for the shopping cart and go to Update shopping cart

#### What are the relevant pivotal tracker/Trello stories?
      * 184651349

### CODE REVIEW

**Review by kagame8732 — DISMISSED** [review]
Working properly!

File: src/controllers/create-product.controller.js [hunk 1]
```diff
@@ -1,53 +1,43 @@
...
 
-    // Check for missing required fields
-    if (!name || !image || !price || !quantity || !type || !exDate) {
-      return res.status(400).json({
-        status: 'error',
-        message: 'Missing required fields',
-      });
+    const { error } = validateProduct(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
```
Comment (didasmbalanya) [inline src/controllers/create-product.controller.js]: move this into a middleware before the controller is called. All validation should be handled by a middleware

File: src/controllers/create-product.controller.js [hunk 2]
```diff
@@ -1,53 +1,43 @@
...
 
-    // Check if quantity is positive
-    if (quantity <= 0) {
-      return res.status(400).json({
-        status: 'error',
-        message: 'Quantity must be a positive number',
+    // Verify if product already exists in the user's collection
+    const productExists = await Product.findOne({
+      where: { name, userId },
+    });
```
Comment (didasmbalanya) [inline src/controllers/create-product.controller.js]: If the product already exists, check the quantity and add the new products being added to the already existing

File: src/controllers/disableAccount.controller.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ const update = (req, res) => {
           message: 'User is disabled successfully.',
         });
       } else {
-        res.send({
+        res.status(500).send({
```
Comment (didasmbalanya) [inline src/controllers/disableAccount.controller.js]: why are you using a status 500? Please research on the correct status code

File: src/controllers/searchProducts.controller.js [hunk 4]
```diff
@@ -28,7 +28,10 @@ const searchProduct = async (req, res) => {
   }
 
   const items = await Product.findAll({ where });
-
+  // if (items.length === 0) {
+  // return res.status(404)
+  //  .send({ message: 'Product not found' });
+  // }
```
Comment (didasmbalanya) [inline src/controllers/searchProducts.controller.js]: Do not push commented code into production

**Review by didasmbalanya — DISMISSED** [review]

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]
well done!

**Review by Raphaela664 — DISMISSED** [review]

**Review by nshutip — DISMISSED** [review]

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]
conflict well resolved!

**Review by JulesNIYOMWUNGERI — APPROVED** [review]

**Review by Raphaela664 — APPROVED** [review]

**Pull request merged**